src/examples/zkapps/hello_world/run.ts

```diff
@@ -1,26 +1,214 @@
+console.log(
+  `Updating state from ${initialState} to 4 with Admin Private Key ...`
+);
+txn = await Mina.transaction(feePayer1, () => {
+  zkAppInstance.update(Field(4), adminPrivateKey);
+  zkAppInstance.sign(zkAppPrivateKey);
+});
+
+try {
+  txn.send().wait();
+
+  currentState = await Mina.getAccount(
+    zkAppAddress
+  ).zkapp?.appState[0].toString();
+
+  if (currentState !== '4') {
+    throw new Error(
+      `Current state of ${currentState} does not match 4 after calling update with 4`
+    );
+  } else {
+    console.log(`Current state succesfully updated to ${currentState}`);
+  }
+} catch (err: any) {
+  throw new Error(`
+    Updating from ${initialState} to 4 failed with error ${err.message}`);
+}
+
+const wrongAdminPrivateKey = PrivateKey.random();
+// attempt to update state with value that satisfies preconditions and incorrect admin private key
+console.log(
+  `Attempting to update state from ${currentState} to 16 with incorrect Admin Private Key ...`
+);
+txn = await Mina.transaction(feePayer1, () => {
+  zkAppInstance.update(Field(16), wrongAdminPrivateKey);
+  zkAppInstance.sign(zkAppPrivateKey);
+});
+
+try {
+  txn.send().wait();
+
+  currentState = await Mina.getAccount(
+    zkAppAddress
+  ).zkapp?.appState[0].toString();
+
+  if (currentState !== '16') {
+    throw new Error(
+      `The current state of ${currentState} does not match 16 after calling update with 16`
+    );
+  }
+} catch (err: any) {
+  console.log(
+    `State correctly was not updated with wrong Admin Private Key. Current state is still ${currentState}`
+  );
+}
+
+// attempt to update state with value that fails precondition x.square().assertEquals(squared).
+try {
+  console.log(
+    `Attempting to update state from ${currentState} to the value that fails precondition of 30 ...`
+  );
+
+  txn = await Mina.transaction(feePayer1, () => {
+    zkAppInstance.update(Field(30), adminPrivateKey);
+    zkAppInstance.sign(zkAppPrivateKey);
+  });
+
+  txn.send().wait();
+
+  currentState = await Mina.getAccount(
+    zkAppAddress
+  ).zkapp?.appState[0].toString();
+
+  if (currentState !== '4') {
+    throw new Error(
+      `State was updated from 4 to ${currentState} which fails the precondition x.square().assertEquals(squared);`
+    );
+  }
+} catch (err: any) {
+  console.log(
+    `State correctly was not updated when fails the precondition x.square().assertEquals(squared). Current state is still ${currentState}`
+  );
+}
+
+// attempt simultaneous “Update” method invocation by different users
+try {
+  console.log(
+    `Attempting to simultaneous update the current state of ${currentState} from multiple users ...`
+  );
+
+  // expected to fail and current state stays at 4
+  txn = await Mina.transaction({ feePayerKey: feePayer1, fee: '10' }, () => {
+    zkAppInstance.update(Field(256), adminPrivateKey);
+    zkAppInstance.sign(zkAppPrivateKey);
+  });
+
+  txn.send();
+
+  currentState = await Mina.getAccount(
```

Comment:
same comments as before on properly checking the failure case. what you do -- throwing a different error after something that should've thrown -- would also properly test for the expected failure, but _only if_ you distinguish between those two errors in the catch block